### DISCUSSION TITLE
Add ability to refresh redirects in next.config.js

### DISCUSSION THREAD

**Discussion by felskov** [vercel/next.js#18731] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

We're developing a site, in which editors frequently add new redirect rules, so we'd like to be able to maintain the list of redirected urls in our CMS and fetch them inside `async redirects()`. The problem with doing so is that it seems this method is only ever triggered when the server is started, and is then never refreshed throughout the lifecycle.

## Describe the solution you'd like

We'd like to be able to configure an interval in which next.js refreshes the configuration of redirects, so that for instance an updated list of redirects is fetched from our CMS on a daily basis.

## Describe alternatives you've considered

Currently we're considering forcing the server to reboot on a daily basis to force it to refresh redirect configuration, but we'd much prefer not having to do so.

**Comment by balazsorban44** [comment]
Do you do per page redirects? https://nextjs.org/blog/next-10#redirect-support Would this help?

You could create a redirect property for that page in your CMS which you check in `getStaticProps`. If it exists, you could redirect to it inside `getStaticProps`.

**Comment by felskov** [comment]
@balazsorban44 Thanks for the reply, we'll look into what you suggest as it seems to support our use case :) I still think that it'd be nice to configure some way to refresh redirects from next.config.js, as that would remove the overhead of looking up redirects inside getStaticProps, but it should be negligible in our case.

**Reply by Antonytm** [reply]
I think it will be nice to have this feature.

**Comment by eFFeeMMe** [comment]
Hello everyone, my company is having the same need as OP.

For SEO reasons, our content editors, who use a CMS, would like to be able to update page slugs on some articles, needing a redirect from the previous slug to the new slug, or maybe even redirect all traffic to those articles to non-article kinds of pages.

e.g.
- from `/guides/how-to-something` to `/guides/doing-something`
- from `/guides/how-to-something` to `/articles/how-i-did-something`

The fact that Next's config is only loaded at Next server boot makes the `redirects` config option a bad fit for our use case.

To people landing here still looking for a solution: to implement "CMS-defined" redirects you can use a Next middleware instead.

It would have been nice to see `redirects`'s caveats explicitly listed in the docs though.
